src/main/java/org/junitpioneer/jupiter/AbstractEntryBasedExtension.java

```diff
@@ -147,18 +138,26 @@ public void afterAll(ExtensionContext context) throws Exception {
 	}
 
 	private void restoreOriginalEntries(ExtensionContext context) {
-		getStore(context).get(BACKUP, EntriesBackup.class).restoreBackup();
+		getStore(context).getOrDefault(getStoreKey(context), EntriesBackup.class, new EntriesBackup()).restoreBackup();
 	}
 
 	private Store getStore(ExtensionContext context) {
 		return context.getStore(Namespace.create(getClass()));
 	}
 
+	private Object getStoreKey(ExtensionContext context) {
+		return context.getDisplayName();
```

Comment:
The `Displayname` is *NOT* unique! this can lead to further collisions!
But each test (and container) has an uniqueid. I think this is the more safer way.

Reply:
Good point, switched to [`ExtensionContext#getUniqueId()`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/extension/ExtensionContext.html#getUniqueId()).